### PR TITLE
[Fix/Inference] Fix format of input prompts and input model  in inference engine

### PR DESCRIPTION
**Pull request by yuehuayingxueluo**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
Fixed errors caused by the input prompts are not in the form of a list and the input model is not on the GPU.
pytest:
image

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/batch_bucket.py [hunk 1]
```diff
@@ -447,3 +447,6 @@ def get_sequence_lengths(self) -> torch.Tensor:
     def fd_inter_tensor(self) -> None:
         assert self.fd_interm_tensor is not None, "fd_interm_tensor is not provided"
         return self.fd_interm_tensor
+
+    def __repr__(self) -> str:
+        return f"(sequences_dict={self._sequences_dict}, " f"is_prompts={self.is_prompts})"
```
Comment (yuanheng-zhao) [inline colossalai/inference/batch_bucket.py:452]: This will error out. Watch out pairs of `""` used.

File: colossalai/inference/core/engine.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def __init__(
         self.tokenizer.pad_token = self.tokenizer.eos_token
         self.generation_config = inference_config.to_generation_config(self.model_config)
         model = model.eval()
+        model = model.cuda()
```
Comment (yuanheng-zhao) [inline colossalai/inference/core/engine.py:60]: Then we might want to remove `.cuda()` in `_shardformer` (duplicated)

File: colossalai/inference/core/engine.py [hunk 3]
```diff
@@ -223,6 +224,9 @@ def add_request(
 
         block_size = self.inference_config.block_size
 
+        if prompts is not None and not isinstance(prompts, list):
```
Comment (CjhHa1) [inline colossalai/inference/core/engine.py:227]: request_id also needs to revise and I suggest using ` if isinstance(prompts,str) and isinstance(requests_id,int):`

File: colossalai/inference/core/engine.py (hunk 2 above)
Comment (yuehuayingxueluo) [reply colossalai/inference/core/engine.py:60]: Done

File: colossalai/inference/core/engine.py (hunk 3 above)
Comment (yuehuayingxueluo) [reply colossalai/inference/core/engine.py:229]: Done

File: colossalai/inference/batch_bucket.py [hunk 4]
```diff
@@ -447,3 +447,6 @@ def get_sequence_lengths(self) -> torch.Tensor:
     def fd_inter_tensor(self) -> None:
         assert self.fd_interm_tensor is not None, "fd_interm_tensor is not provided"
         return self.fd_interm_tensor
+
+    def __repr__(self) -> str:
+        return f"(sequences_dict={self._sequences_dict}, is_prompts={self.is_prompts})"
```
Comment (yuanheng-zhao) [inline colossalai/inference/batch_bucket.py:452]: Is it supposed to be used only in testing?

**Review by yuanheng-zhao — APPROVED** [review]

**Review by CjhHa1 — APPROVED** [review]
LGTM

File: colossalai/inference/batch_bucket.py (hunk 4 above)
Comment (yuehuayingxueluo) [reply colossalai/inference/batch_bucket.py:452]: I have added logger.debug in test_batch_bucket.py

**Pull request merged**